### ISSUE TITLE
Not able to set multiple mosquitto listens

### ISSUE THREAD

**Issue by Vishal000** [eclipse/mosquitto#1666]
I am setting the following in my mosquitto.conf file:
password_file /etc/mosquitto/passwd
allow_anonymous false
log_dest none
max_inflight_messages 1
set_tcp_nodelay true
bind_address localhost
socket_domain ipv4

Environment: Yocto Linux 2.4
Mosquitto version: 1.5.8

In the above mosquitto.conf file, I am not able to set both the bind_address and the socket_domain together. Only one will work. Both will not work simultaneously. Can anyone guide here ?

Basically, I need to restrict the MQTT packets to localhost and IPv4 only.
Please help here.

**Comment by ralight** [comment]
First suggestion - try it without logging turned off (remove `log_dest none`) to see what error it gives.

Second suggestion, try:

```
listener 1883 127.0.0.1
socket_domain ipv4
```

Third suggestion: Do you really need to restrict it to IPv4 only? If IPv6 isn't configured, then it just won't open a listener for IPv6.

**Comment by Vishal000** [comment]
I tried out your second suggestion. But the MQTT packets are not restricted to the localhost. Does it ?

**Comment by Vishal000** [comment]
The main reason to restrict to ipv4 is that since ipv6 is not used, the mosquitto broker restarts automatically after some time. So, I wanted to check if the restriction to ipv4 works. 
Do you know any specific fix ?

**Comment by karlp** [comment]
you might perhaps want to look into the actual cause of "if ipv6 is not used, the broker restarts automatically..."

**Comment by ralight** [comment]
Did you try suggestion one? Seeing the logs would be helpful.

For suggestion two, if you used that as your entire config file, then there is something seriously wrong. There is essentially no way to start a listener with that configuration and having it not bind to localhost. Could you please run `netstat -atln` and post the output when mosquitto is running with that configuration?

**Comment by Vishal000** [comment]
![image](https://user-images.githubusercontent.com/5052727/80445156-2ecfe200-8931-11ea-9f77-fc62bdc3243b.png)

My entire configuration file is:
password_file /etc/mosquitto/passwd
allow_anonymous false
max_inflight_messages 1
set_tcp_nodelay true
listener 1883 127.0.0.1
socket_domain ipv4

The broker seems to be working. Do you spot any issues here ?

**Comment by Vishal000** [comment]
@karlp : I have searched a lot, but did not find the any solution. Can you help here ?

**Comment by Vishal000** [comment]
@ralight : 
![image](https://user-images.githubusercontent.com/5052727/80447532-41e5b080-8937-11ea-8b3b-034a0cd29f46.png)

Yes, I tried the suggestion one. You can see the mosquitto.conf file and the error it gives.

**Comment by ralight** [comment]
`tcp        0      0 127.0.0.1:1883             0.0.0.0:*               LISTEN`

This line means that the socket listening on port 1883 is bound to 127.0.0.1, i.e. the loopback interface. It is not possible to connect to that device using the 127.0.0.1 address from a remote computer. The configuration is working.

**Comment by Vishal000** [comment]
Yeah, thank you @ralight .
